services/lobby.service.js

```diff
@@ -0,0 +1,1003 @@
+    }
+};
+
+
+const playLobby = async (req, res) => {
+    try {
+        const { id } = req.body;
+
+        // Check if the user exists
+        const user = await User.findOne({ _id: id });
+        if (!user) {
+            return Constants.USER_NOT_FOUND;
+        }
+
+        // Check if the user has already created a lobby
+        const lobby = await Lobby.findOne({ lobbyCreator: id });
+        if (!lobby) {
+            return Constants.CREATE_LOBBY_FIRST;
+        }
+
+        // Check if the lobby has at least 3 players
+        if (lobby.playerList.length < 3) {
+            return Constants.MINIMUM_PLAYERS_REQUIRED;
+        }
+
+        if (lobby?.rounds[0]?.loop1?.length) {
+            return "mafi mil ski ha"
+        }
+
+        // Update the user's canPlayGame status
+        await User.findByIdAndUpdate(id, { canPlayGame: true }, { new: true });
+
+        // Lock the lobby and start the round
+        const updatedLobby = await Lobby.findByIdAndUpdate(
+            lobby._id.toString(),
+            { lobbyLocked: true },
+            { new: true }
+        );
+
+        // Start the round and get the updated lobby data
+        const result = await startRound(updatedLobby);
+        return {
+            code: 'success',
+            message: Constants.GAME_PLAYED_SUCCESSFULLY,
+            lobby: result,
+        };
+    } catch (error) {
+        console.log(error);
+        return Constants.INTERNAL_SEVER_ERROR;
+    }
+};
+
+
+
+const statusLobby = async (req, res) => {
+    try {
+        const { lobbyId, userId, status } = req.body;
+        const lobby = await Lobby.findOne({ _id: lobbyId });
+
+        if (!lobby) {
+            return 'Lobby not found.';
+        }
+
+        if (!status && lobby.playerList.includes(userId)) {
+            // Remove userId from the playerList using array filtering
+            lobby.playerList = lobby.playerList.filter((player) => player !== userId);
+
+            if (lobby.lobbyCreator.toString() === userId) {
+                lobby.lobbyCreator = lobby.playerList[0];
+            }
+
+            await lobby.save();
+        }
+
+        return lobby;
+
+    } catch (error) {
+        console.log(error, 'error')
+        return 'Internal server error.';
+    }
+};
+
+
+const updateRound = async (_id, round) => {
+    const result = await Lobby.findByIdAndUpdate(
+        _id,
+        { $push: { rounds: { round: round, questions: [] } } },
+        { new: true }
+    )
+    console.log(result, 'asdasdasdas')
+    return result
+}
+
+
+const startRound = async (req, res) => {
+    try {
+        const { body } = req;
+        if (body) {
+            console.log(body, 'body');
+            let findLobby = await Lobby.findOne({ lobbyCode: body.lobbyCode });
+            console.log('11111111111111111111');
+            if (body.round === '0') {
+                return 'Lobby match not start yet.'
+            }
+            else if (body.round === '1') {
```

Comment:
Let's create the following enum to capture round numbers instead of using hardcoded strings

`
const RoundNumber = {
  FIRST: '1',
  SECOND: '2',
  THIRD: '3',
};
`